xbmc/peripherals/addons/PeripheralAddon.cpp

```diff
@@ -71,7 +72,20 @@ CPeripheralAddon::~CPeripheralAddon(void)
 {
   // delete all peripherals provided by this addon
   for (const auto& peripheral : m_peripherals)
+  {
+    if (CSettings::GetInstance().GetBool(CSettings::SETTING_INPUT_CONTROLLERPOWEROFF))
+    {
+      // shutdown the joystick if it is supported
+      if (peripheral.second->Type() == PERIPHERAL_JOYSTICK)
+      {
+        CPeripheralJoystick* joystick = static_cast<CPeripheralJoystick*>(peripheral.second);
+        if (joystick->SupportsPowerOff())
+          PowerOffJoystick(joystick->RequestedPort());
```

Comment:
Should be `PowerOffJoystick(peripheral.first);` - RequestedPort() can be -1. Will fix and squash into PR